cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -232,7 +232,11 @@ set(EMSCRIPTEN_SYSROOT "${_emcache_output}/sysroot")
 list(APPEND CMAKE_FIND_ROOT_PATH "${EMSCRIPTEN_SYSROOT}")
 list(APPEND CMAKE_SYSTEM_PREFIX_PATH /)
 
-set(CMAKE_LIBRARY_ARCHITECTURE "wasm32-emscripten")
+if ($ENV{CFLAGS} MATCHES "MEMORY64")
```

Comment:
I'm not that familiar with CMake, but wouldn't the environment var CFLAGS have something like `-sMEMORY64`?

Reply:
MATCHES will match any string contains `MEMORY64` so it will match `-s MEMORY64` and `-sMEMORY64` and `-sMEMORY64=2` and `-s MEMORY64=2`.

It will also match `MEMORY64=0` which is a little unfortunate... but I'm not sure we need to worry too much about that edge case.